src/storage/src/source/postgres/replication.rs

```diff
@@ -220,7 +220,15 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                 // If the publication gets deleted there is nothing else to do. These errors
                 // are not retractable.
                 for &oid in table_info.keys() {
-                    let update = ((oid, Err(err.clone())), *data_cap.time(), 1);
+                    // Produce definite errors at the maximum LSN because we
+                    // need to ensure that the update cannot be ignored by the
+                    // persist sink, which ignores any data produced at times
+                    // less than the write frontier. The `resume_lsn` isn't
+                    // appropriate to produce this data because individual
+                    // subsource frontiers may be in advance of the
+                    // `resume_lsn`, in which case this error would not be
+                    // written.
```

Comment:
I think the choice of LSN should be justified on account of definiteness and not of persist-sink. `persist-sink` has that behavior *because* it assumes definiteness and an alternative persist-sink implementation could be validating that the supplied data is identical for the times that are already written.

So something like:

> We must emit this error at a definite LSN which ideally would be the LSN at which the `DROP PUBLICATION` action was written to in the upstream database. Unfortunately we don't have a way to learn that LSN and so we must choose an LSN out of thin air that is guaranteed to not invalidate the definite decisions previously made by this operator. We therefore always pick `u64::MAX`, which will(in practice) never conflict any previously revealed portions of the TVC.